3.11 Tests Don't Run

### Please make sure these conditions are met

- [x] I have checked that this issue has not already been reported.
- [ ] I have confirmed this bug exists on the latest version of scanpy.
- [x] (optional) I have confirmed this bug exists on the main branch of scanpy.

### What happened?

See https://github.com/scverse/integration-testing/actions/runs/13409148582/job/37455927886?pr=15

I was able to reproduce locally with a fresh python 3.11 environment

### Minimal code sample

```python
pytest
```

### Error output

```pytb
============================================================== ERRORS ===============================================================
______________________________________________ ERROR collecting src/scanpy/_compat.py _______________________________________________
../../../miniconda3/lib/python3.12/doctest.py:948: in find
    self._find(tests, obj, name, module, source_lines, globs, {})
../../../miniconda3/lib/python3.12/doctest.py:1022: in _find
    self._find(tests, val, valname, module, source_lines,
../../../miniconda3/lib/python3.12/doctest.py:1010: in _find
    test = self._get_test(obj, name, module, globs, source_lines)
../../../miniconda3/lib/python3.12/doctest.py:1078: in _get_test
    lineno = self._find_lineno(obj, source_lines)
../../../miniconda3/lib/python3.12/doctest.py:1127: in _find_lineno
    obj = inspect.unwrap(obj).__code__
E   AttributeError: 'method_descriptor' object has no attribute '__code__'. Did you mean: '__call__'?
====================================================== short test summary info ======================================================
ERROR src/scanpy/_compat.py - AttributeError: 'method_descriptor' object has no attribute '__code__'. Did you mean: '__call__'?
```

### Versions

<details>

```
typing_extensions       4.12.2
toolz   1.0.0
legacy-api-wrap 1.4.1
pytz    2025.1
PyYAML  6.0.2
leidenalg       0.10.2
importlib_metadata      8.6.1
six     1.17.0
pillow  11.1.0
natsort 8.4.0
llvmlite        0.44.0
threadpoolctl   3.5.0
numba   0.61.0
scikit-learn    1.5.2
kiwisolver      1.4.8
anndata 0.11.3
igraph  0.11.8
cycler  0.12.1
zarr    2.18.4
numcodecs       0.15.1
session-info2   0.1.2
cloudpickle     3.1.1
pandas  2.2.3
python-dateutil 2.9.0.post0
setuptools      75.8.0
texttable       1.7.0
joblib  1.4.2
packaging       24.2
h5py    3.13.0
numpy   2.1.3
zipp    3.21.0
Deprecated      1.2.18
wrapt   1.17.2
coverage        7.6.12
pyparsing       3.2.1
asciitree       0.3.3
dask    2024.7.1
matplotlib      3.10.0
scipy   1.15.2
----    ----
Python  3.11.9 (main, Apr  2 2024, 08:25:04) [Clang 15.0.0 (clang-1500.1.0.2.5)]
OS      macOS-15.1-arm64-arm-64bit
CPU     8 logical CPU cores, arm
GPU     No GPU found
Updated 2025-02-19 11:48
```

</details>

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -111,7 +111,8 @@ test-full = [
     "scanpy[dask-ml]",
 ]
 doc = [
-    "sphinx>=7",
+    # https://github.com/sphinx-doc/sphinx/issues/13366
+    "sphinx>=7,<8.2.0",
     "sphinx-book-theme>=1.1.0",
     "scanpydoc>=0.14.1",
     "sphinx-autodoc-typehints>=1.25.2",

--- a/src/scanpy/_compat.py
+++ b/src/scanpy/_compat.py
@@ -108,7 +108,6 @@ else:
 
 if sys.version_info >= (3, 11):
 
-    @wraps(BaseException.add_note)
     def add_note(exc: BaseException, note: str) -> None:
         exc.add_note(note)
 else:

